### PR TITLE
BUG: DataFrame.apply axis=1 for str ops with no axis argument

### LINKED ISSUE #39211: BUG: DataFrame.apply axis=1 for str ops with no axis argument
E.g.

```
df = pd.DataFrame({'a': [1, 2], 'b': [3, 4]})
print(df.apply('pct_change', axis=0))
print(df.apply('pct_change', axis=1))
```

both give the output

```
     a         b
0  NaN       NaN
1  1.0  0.333333
```

This should instead raise rather than silently give incorrect results.

### PR DESCRIPTION
**Pull request by rhshadrach**
- [x] closes #39211
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

**Review by erfannariman — APPROVED** [review]

File: pandas/tests/frame/apply/test_frame_apply.py [hunk 1]
```diff
@@ -179,6 +179,16 @@ def test_apply_with_string_funcs(self, request, float_frame, func, args, kwds, h
         expected = getattr(float_frame, func)(*args, **kwds)
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "how, args", [("pct_change", ()), ("nsmallest", (1, ["a", "b"])), ("tail", 1)]
+    )
```
Comment (jreback) [inline pandas/tests/frame/apply/test_frame_apply.py:184]: does this already raise when a function is passed? ideally can you co-locate these tests.

Comment (rhshadrach) [reply pandas/tests/frame/apply/test_frame_apply.py:184]: When function is passed, apply with axis=1 computes the correct result. E.g.

```
df = DataFrame([[1, 2, 3], [4, 5, 6]])
print(df.apply(lambda x: x.tail(1), axis=1))

# gives:
   2
0  3
1  6
```

The argument `axis=1` is intercepted and then `apply_standard` is computed on the rows of the frame without an axis argument.

Comment (jreback) [reply pandas/tests/frame/apply/test_frame_apply.py:184]: ok cool, can you co-locate this test near the others

Comment (rhshadrach) [reply pandas/tests/frame/apply/test_frame_apply.py:184]: What is meant by others? Currently located below the other test for str arguments. Tests for axis=1 and tests that raise are fairly spread out in this file, not sure how else to locate this test.

As an aside, I've been wanting to consolidate (and then start to cleanup) the apply/agg/transform tests into `pandas.tests.apply`. Does that move make sense?

Comment (jreback) [reply pandas/tests/frame/apply/test_frame_apply.py:184]: ok cool
yeah would love to break up some of these test files more logically

**Comment by jreback** [comment]
thanks @rhshadrach pls create an issue (or PR!) for re-orging the apply tests :->

**Pull request merged**